src/Orleans/LogConsistency/IProtocolServices.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
+        /// </summary>
+        /// <returns></returns>
+        string MyClusterId { get; }
+
+    
+        /// <summary>
+        /// The current multicluster configuration of this silo 
+        /// (as injected by the administrator) or null if none.
+        /// </summary>
+        MultiClusterConfiguration MultiClusterConfiguration { get; }
+
+        /// <summary>
+        /// List of all clusters that currently appear to have at least one active
+        /// gateway reporting to the multi-cluster network. 
+        /// There are no guarantees that this membership view is complete or consistent.
+        /// If there is no multi-cluster network, returns a list containing the single element "I".
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<string>  ActiveClusters { get; }
+
+
+        void SubscribeToMultiClusterConfigurationChanges();
+
+        void UnsubscribeFromMultiClusterConfigurationChanges();
+
+
+        #region Logging Functionality
+
+        /// <summary>
+        /// Log an error that occurred in a log-consistency protocol.
+        /// </summary>
+        void ProtocolError(string msg, bool throwexception);
+
+        /// <summary>
+        /// Log an exception that was caught in the log-consistency protocol.
+        /// </summary> 
+        void CaughtException(string where, Exception e);
+
+        /// <summary>
+        /// Log an exception that occurred in user code, for some callback
+        /// </summary>
+        /// <param name="callback">The name of the callback</param>
+        /// <param name="where">The context from which the callback was called</param>
+        /// <param name="e">The caught exception</param>
+        void CaughtUserCodeException(string callback, string where, Exception e);
+
+        /// <summary> Output the specified message at <c>Info</c> log level. </summary>
+        void Info(string format, params object[] args);        
+        /// <summary> Output the specified message at <c>Verbose</c> log level. </summary>
+        void Verbose(string format, params object[] args);
+        /// <summary> Output the specified message at <c>Verbose2</c> log level. </summary>
+        void Verbose2(string format, params object[] args);
+        /// <summary> Output the specified message at <c>Verbose3</c> log level. </summary>
+        void Verbose3(string format, params object[] args);
```

Comment:
Hmm, this did not really work... ErrorCodes are internal to runtime, so I can't format errors from outside the runtime. I don't want to create a dependency from OrleansEventSourcing.dll on OrleansRuntime.dll. And the error reporting should use the same formatting as the logging. So, for now I will keep this as it is mostly, and only remove the verbosity overloads.